active/NavigationView/HierarchicalNav/HierarchicalNavigationView.md

```diff
@@ -0,0 +1,265 @@
+            }
+        }
+```
+
+# Remarks
+<!-- Explanation and guidance that doesn't fit into the Examples
+section.  For example, see the Remarks for the MediaPlayerElement 
+(https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.MediaPlayerElement#remarks). -->
+
+
+# API Notes
+<!-- Give a one or two line description of each API (type
+and member), or at least the ones that aren't obvious
+from their name.  These descriptions are what show up
+in IntelliSense. -->
+
+
+# API Details
+```c++
+[WUXC_VERSION_PREVIEW]
+[webhosthidden]
+runtimeclass NavigationViewExpandingEventArgs
+{
+    NavigationViewItemBase ExpandingItemContainer { get; };
+}
+
+[WUXC_VERSION_PREVIEW]
+[webhosthidden]
+runtimeclass NavigationViewCollapsedEventArgs
+{
+    NavigationViewItemBase CollapsedItemContainer { get; };
+}
+
+[WUXC_VERSION_RS3]
+[webhosthidden]
+[WUXC_INTERFACE_NAME("INavigationView", f209ce15-391a-42ca-9fc6-f79da65aca32)]
+[WUXC_STATIC_NAME("INavigationViewStatics", 363a86c7-72da-4420-b871-15d9d0d45756)]
+[WUXC_CONSTRUCTOR_NAME("INavigationViewFactory", e50687c1-b7c2-4975-ad7a-5f4fe6a514c9)]
+[MUX_PROPERTY_CHANGED_CALLBACK(TRUE)]
+[MUX_PROPERTY_CHANGED_CALLBACK_METHODNAME("OnPropertyChanged")]
+unsealed runtimeclass NavigationView : Windows.UI.Xaml.Controls.ContentControl
+{
+...
+    [WUXC_VERSION_PREVIEW]
+    {
+        event Windows.Foundation.TypedEventHandler<NavigationView, NavigationViewExpandingEventArgs> Expanding;
+        event Windows.Foundation.TypedEventHandler<NavigationView, NavigationViewCollapsedEventArgs> Collapsed;
+
+        void Expand(NavigationViewItem item);
+        void Collapse(NavigationViewItem item);
+    }
+}
+
+[WUXC_VERSION_RS3]
+[webhosthidden]
+[WUXC_INTERFACE_NAME("INavigationViewItemBase", edf04eb1-37d1-471f-8570-3829ee5b2bc6)]
+[WUXC_CONSTRUCTOR_NAME("INavigationViewItemBaseFactory", eb014cef-7890-4ebb-8245-02e8510f321d)]
+[default_interface]
+unsealed runtimeclass NavigationViewItemBase : Windows.UI.Xaml.Controls.ContentControl
+{
+    [WUXC_VERSION_PREVIEW]
+    {
+        Boolean IsSelected{ get; set; };
+        static Windows.UI.Xaml.DependencyProperty IsSelectedProperty { get; };
+    }
+}
+
+[WUXC_VERSION_RS3]
+[webhosthidden]
+[WUXC_INTERFACE_NAME("INavigationViewItem", 8614be0f-b7b6-4851-960a-f5e3f69f624a)]
+[WUXC_STATIC_NAME("INavigationViewItemStatics", 803c0081-fda5-4b90-aace-3f2306dbe5c4)]
+[WUXC_CONSTRUCTOR_NAME("INavigationViewItemFactory", 973bdb4a-7e08-4f76-923c-f12bd685e86e)]
+unsealed runtimeclass NavigationViewItem : NavigationViewItemBase
+{
+...
+    [WUXC_VERSION_PREVIEW]
+    {
+        [MUX_DEFAULT_VALUE("false")]
+        Boolean IsExpanded{ get; set; };
+        [MUX_DEFAULT_VALUE("false")]
+        Boolean HasUnrealizedChildren{ get; set; };
+        [MUX_DEFAULT_VALUE("false")]
+        Boolean IsChildSelected{ get; };
+
+        [MUX_PROPERTY_CHANGED_CALLBACK(TRUE)]
+        Windows.Foundation.Collections.IVector<Object> MenuItems{ get; };
```

Comment:
we could make this content property for NavigationViewItem

Reply:
As Dilip mentioned earlier, NavigationViewItem already has a Content property because it derives from ContentControl. Today, this property is already used to set the text label, or other content, to the right of the icon.

Reply:
@ojhad is this still true after the rewrite with ItemsRepeater?
> it derives from ContentControl

<br/>
(Are NavigationViewItemBase objects still ListViewItems?)

Reply:
Found the answer in the idl. NavigationViewItemBase now derives directly from ContentControl and not ListViewItem.

Reply:
So that's still a problem for [ContentProperty].

And if this is a breaking change, don't we have to call this 3.0? (And then DCPP becomes 4.0?)

Reply:
I don't think this is a breaking change, I think we want to add a new MenuItems property. My only question is whether MenuItems should be added to NavigationViewItemBase or to NavigationViewItem?

Reply:
I see. This is not a breaking change. The MenuItems is a new property.

Reply:
It's already there: [NavigationView.MenuItems](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.NavigationView.MenuItems)

Reply:
Are we talking about different propertes ? I was talking about the new MenuItems property we are adding on NavigationViewItem (not NavigationView, which already exisits) to get the nesting support.